kernel/src/engine/mod.rs

```diff
@@ -5,26 +5,32 @@
 #[cfg(feature = "arrow-conversion")]
 pub(crate) mod arrow_conversion;
 
-#[cfg(feature = "arrow-expression")]
+#[cfg(all(
```

Comment:
This was broken before because `arrow_expression` uses stuff in `arrow_utils` so enabling _only_ `arrow_expression` would not have compiled.